backend/index.js

```diff
@@ -40,18 +40,76 @@ app.post('/create', async (req, res) => {
   }
 });
 
-
-app.get('/', (req, res) => {
-  res.send('Welcome to the Express server!');
-});
-
-app.post('/submit', (req, res) => {
+/**
+ * post /submit
+ * 
+ * Submits text to firebase under texts collection.
+ * 
+ * Responses:
+ * - 200: Returns success message
+ */
+app.post('/submit/text', (req, res) => {
   const { inputText } = req.body; 
   console.log('Received input:', inputText);
   res.status(200).json({ message: `Form data received successfully: ${inputText}` });
   createText(inputText)
 });
 
+/**
+ * GET /api/data/questions
+ * 
+ * Retrieves data from firebase from questions collection.
+ * 
+ * Responses:
+ * - 200: Returns an array of data matching the query parameters.
+ * - 500: Server error if something goes wrong while fetching data.
+ */
+app.get('/api/data/questions', async (req, res) => {
+  try {
+    const snapshot = await db.collection("questions").get();
+    const data = [];
+
+    if (snapshot.empty) {
+      console.log('No matching documents.');
+      return;
+    }  
+
+    snapshot.forEach((doc) => {
+      console.log(doc.id, '=>', doc.data());
+      data.push({ id: doc.id, ...doc.data() });
+    });
+    res.status(200).json(data);
+  } catch (error) {
+    console.error('Error fetching data from Firebase:', error);
+    res.status(500).json({ message: 'Error fetching data from Firebase' });
+  }
+});
+
+/**
+ * POST /createQ
+ * 
+ * Creates questions from form data and store in firebase
+ * 
+ * Responses:
+ * - 500: Server error if something goes wrong while fetching data.
+ */
+app.post('/createQ', async (req, res) => {
+  try {
+      console.log(req.body);
+      const questionJson = {
+          title: req.body.title,
+          category: req.body.category,
+          complexity: req.body.complexity ,
+          description: req.body.description
+      };
+      const response = db.collection("questions").doc().set(questionJson); // Added 'await'
+      res.send({ message: "Question created successfully", response });
+  } catch (error) {
+      res.status(500).send({ error: error.message });
+  }
+});
```

Comment:
inconsistent indentation compared to rest of file (4-space tab here compared to 2-space tab in rest of file)